### PR TITLE
[BUGFIX] Add error handling to CLI get a default suite name

### PR DESCRIPTION
**Pull request by luke321321**
Changes proposed in this pull request:
- Add error handling to a CLI method to get a default suite name.  This method can sometimes be passed incomplete `BatchRequest` information.  This leads to a `TypeError` as shown in screenshot from CLI.  Added a basic unit test for this behaviour.

Screenshot 2023-02-20 at 16 36 17

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/cli/test_toolkit.py [hunk 1]
```diff
@@ -524,3 +524,15 @@ def test_get_relative_path_from_config_file_to_data_base_file_path_from_misc_dir
...
     obs = get_relative_path_from_config_file_to_base_path(ge_dir, absolute_path)
     assert obs == os.path.join("..", "..", "data", "pipeline1")
+
+
+def test_get_default_expectation_suite_name_with_partial_batch_request_definition():
+    batch_request = {
+        "datasource_name": "snowflake",
+        "data_connector_name": "default_runtime_data_connector_name",
+        "limit": 1000,
+    }
```
Comment (luke321321) [inline tests/cli/test_toolkit.py:534]: This is the exact `batch_request` that is passed to `get_default_expectation_suite_name` from the CLI error:

Screenshot 2023-02-20 at 16 36 17

**Comment by luke321321** [comment]
This is related to #5462 but is fixing the error and not the root cause of the error -- that there is no `data_asset_name` for a `RuntimeDataConnector` given by the CLI

**Comment by Shinnnyshinshin** [comment]
This is a fantastic way to address the issue. Thank you for the contribution @luke321321

**Review by Shinnnyshinshin — APPROVED** [review]
✅

**Comment by Shinnnyshinshin** [comment]
Actually @luke321321, I think this PR and #7187 are addressing similar issues. Do you have a preference on which way you would like to do the fix?

**Comment by luke321321** [comment]
@Shinnnyshinshin : #7187 is a better fix as addresses the root cause so is more of a priority.  

There's still the possibility of `batch_request` that is passed to this method is unable to create an instance of `BatchRequest` as there's no validation on the `batch_request` variable.  All we're wanting the `BatchRequest` is for an `id`, so there's an overhead/risk to create the object with no error handling  So in my opinion there's still the scope for this PR but the immedidate need is taken case of #7187.

I'm happy to leave it up to you if you just wanted to merge #7187 and close this PR.

**Comment by Shinnnyshinshin** [comment]
Thank you @luke321321 I think #7187 will be the way to go. Will close this PR as you mentioned.

**Pull request closed without merging**